app/src/main/java/vn/sunasterisk/music_70/ui/discover/DiscoverFragment.kt

```diff
@@ -59,8 +60,14 @@ class DiscoverFragment : BaseFragment(), GenreContract.View {
         genrePresent.getTrack(apiTrending, true)
 
         listSongAdapter = ListSongAdapter({
-            val index= listSongAdapter.getListData().indexOf(it)
-            startActivity(NowPlayingActivity.getIntent(requireContext(), listSongAdapter.getListData(),index))
+            val index = listSongAdapter.getListData().indexOf(it)
```

Comment:
Bạn viết thêm 1 phương thức onClickSongAt(position: Int) để thực hiện các việc này.
Rồi truyền vào như sau:
```suggestion
            listSongAdapter = ListSongAdapter(::onClickSongAt) {
            }
```

Reply:
Ở `ListSongAdapter `ông có cài đặt phương thức `onItemClick ` truyền vào Track,
dẫn tới khi callback về Fragment ông lại phải tìm index và lấy lại list đó 
Nếu như ViewHolder của ông là inner class thì tại sao ông không truyền thằng listData và index qua hàm onItemClick 
`class ListSongAdapter(private val onItemClicked: (track: Track) -> Unit, private val optionClick: (track: Track) -> Unit) :
    RecyclerView.Adapter<ListSongAdapter.ViewHolder>() {`
```suggestion
 class ListSongAdapter(
     private val onItemClicked: (tracks: List<Track>, index: Int) -> Unit, 
     private val optionClick: (track: Track) -> Unit
 ) : RecyclerView.Adapter<ListSongAdapter.ViewHolder>() {
```